### PR TITLE
fix support for elasticsearch v7 + add types

### PR DESCRIPTION
**Pull request by robyte-ctrl**
opened the PR in here by mistake (meant to merge it in my fork's `master` branch) so feel free to close it

but this PR does fix https://github.com/pinojs/pino-elasticsearch/issues/166 and it also adds types so it's easier to use this library in a typescript project

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Good work!

This is a semver-major change, right?

**Comment by robyte-ctrl** [comment]
> Good work!
> 
> This is a semver-major change, right?

@mcollina shouldn't be, although I have renamed some of the options so that they follow the camelCase naming notation, the old names still work and I've added some tests to make sure that they do

if you want to remove the old names completely then that would be a semver-major change

**Comment by mcollina** [comment]
I see you have removed --bulk-size from the CLI options, hence my guess.

File: lib.d.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import type { Transform } from 'stream'
```
[the remark this answers is not in the document]
Comment (robyte-ctrl) [reply lib.d.ts:1]: done

though it's first time I tried tests for types so I'm not sure how good they are

**Pull request merged**